message_passing/update_with_start/lazy_initialization/starter.py

```diff
@@ -0,0 +1,68 @@
+import asyncio
+from typing import Optional, Tuple
+
+from temporalio import common
+from temporalio.client import (
+    Client,
+    WithStartWorkflowOperation,
+    WorkflowHandle,
+    WorkflowUpdateFailedError,
+)
+
+from message_passing.update_with_start.lazy_initialization.workflows import (
+    ShoppingCartItem,
+    ShoppingCartWorkflow,
+)
+
+
+async def handle_add_item_request(
+    session_id: str, item_id: str, quantity: int, temporal_client: Client
+) -> Tuple[Optional[int], WorkflowHandle]:
+    """
+    Handle a client request to add an item to the shopping cart. The user is not logged in, but a session ID is
+    available from a cookie, and we use this as the cart ID. The Temporal client was created at service-start
+    time and is shared by all request handlers.
+
+    A Workflow Type exists that can be used to represent a shopping cart. The method uses update-with-start to
+    add an item to the shopping cart, creating the cart if it doesn't already exist.
+
+    Note that the workflow handle is available, even if the Update fails.
+    """
+    cart_id = f"cart-{session_id}"
+    start_op = WithStartWorkflowOperation(
+        ShoppingCartWorkflow.run,
+        id=cart_id,
+        id_conflict_policy=common.WorkflowIDConflictPolicy.USE_EXISTING,
+        task_queue="uws",
+    )
+    try:
+        price = await temporal_client.execute_update_with_start_workflow(
+            ShoppingCartWorkflow.add_item,
+            ShoppingCartItem(sku=item_id, quantity=quantity),
+            start_workflow_operation=start_op,
+        )
+    except WorkflowUpdateFailedError:
+        price = None
```

Comment:
May want a bit more of a conditional to confirm it's the failure you expect instead of swallowing every update failure ever. In .NET I checked the application failure's error type: https://github.com/temporalio/samples-dotnet/blob/86a49887aaa668beb3cadb4fa862574f3295fcc6/src/UpdateWithStartLazyInit/Program.cs#L85-L90